src/plugins/core/borders.ts

```diff
@@ -219,6 +219,15 @@ export class BordersPlugin extends CorePlugin<BordersPluginState> implements Bor
     return Object.keys(sheetBorders).map((index) => parseInt(index, 10));
   }
 
+  /**
+   * Get all the rows which contains at least a border
+   */
+  private getRowsWithBorders(sheetId: UID): number[] {
+    const sheetBorders = this.borders[sheetId]?.filter(isDefined);
+    if (!sheetBorders) return [];
+    return Object.keys(sheetBorders[0]).map((index) => parseInt(index, 10));
```

Comment:
I'm afraid  this fix is not enough :(
If you add borders on the last row but *not in the first column* (`sheetBorders[0]`), the same issue is still there

Reply:
Good catch!